lang/rust/avro/src/reader.rs

```diff
@@ -804,6 +804,38 @@ mod tests {
         assert_eq!(expected_value, val);
     }
 
+    #[test]
+    fn test_avro_3507_single_object_reader_incomplete_reads() {
+        let obj = TestSingleObjectReader {
+            a: 42,
+            b: 3.33,
+            c: vec!["cat".into(), "dog".into()],
+        };
+        let mut to_read_1 = Vec::<u8>::new();
+        to_read_1.extend_from_slice(&[0xC3, 0x01]);
```

Comment:
do we need the Vec+extend_from_slice() ? Can't we just do `let to_read_1 = vec!(0xC3, 0x01);` ?
Also please add a comment that those two bytes are the special magic ones.

Reply:
done